TACC/Core-Portal
# Task/FP-997:  improve apps, add small queue support and update execution systems when needed

## Overview: ##

This PR works to improve apps and support 'small' queue:
* Update execution systems when needed
* Support 'small' (and updated 'normal') queue
* Improve app validation

## Related Jira tickets: ##

* [FP-997](https://jira.tacc.utexas.edu/browse/FP-997)
* [FP-1041](https://jira.tacc.utexas.edu/browse/FP-1041)
* [FP-1042](https://jira.tacc.utexas.edu/browse/FP-1042)

## Summary of Changes: ##

- Keep execution system clone in sync with revision of target
- Disable submit when form is invalid (note: no longer include Warning (!) icon for validation errors)
- Handle case when queue's maxNodes is -1
- No longer show warning icon for validation errors 
- Handle normal/small queue during node count validation
- Check for normal queue trying to use a serial app 
- Validate form on mount to show error immediately
- Bump formik due to getting initial validation to work (https://github.com/formium/formik/issues/2639#issuecomment-663623448, https://github.com/formium/formik/issues/2368, https://github.com/formium/formik/issues/2624)

## Testing Steps: ##

1. Select an app
2. See that validation occurs immediately

------ 

1. Select an app
2. Select small queue
3. Select Node Count to be 3 or greater and ensure there is a validation error 

-----

1. Select an app
2. Select normal queue
3. Select Node Count to be less than 3 and ensure there is a validation error 


-----
1. Select a SERIAL app like Frontera HPC Jupyter
2. Select normal queue
3. Ensure there is a validation error 
-----

1. Select a SERIAL app like Frontera HPC Jupyter
2. Run using rtx queue (_to ensure we handle case when queue's maxProcessorsPerNode is -1_)

## Notes: ##
* The clonedSystemId is based on login-host, execution type, user name and (now added in this PR) revision. However, it is not unique enough if we have multiple execution systems defined for a system. Currently, we have only 2 execution systems (one for each execution type). It is possible that we have multiple execution systems in the future. As we are limited in name length, should we add a short hash of the original execution system to our cloned system name before the revision: like nathanf.TACC-ACI.exec.frontera.HPC.HASH.REVISION. This is not done in this PR but possibly something to note for follow work (ticket? @rstijerina  @xandroc ).
* https://bitbucket.org/taccaci/portal-agave-templates/pull-requests/13/fp-997-add-small-queue contains the updated tapis execution system definition


## TODO

- [x] backend tests (missing; created follow on Jira ticket https://jira.tacc.utexas.edu/browse/FP-1107)
- [x] frontend tests for yup validation
- [x] any other frontend tests
- [x] remove parameters+inputs from initialTouched ~~why isn't email field showing validation for Frontera Jupyter app on initial load of app (TODO: confirm that this is even desired)~~. **Note from Jul 5 dev meeting discussion: add any parameters/inputs that have a default value to initialTouched.** 
- [x] remove normal queue from dropdown if SERIAL app
- [x] fix incorrect validation error that appears after job submission
- [ ] Update execution system to ensure `small` is the default queue.
